api/keystore/service.go

```diff
@@ -266,8 +266,12 @@ func (ks *Keystore) ImportUser(r *http.Request, args *ImportUserArgs, reply *Imp
 
 	ks.log.Verbo("ImportUser called for %s", args.Username)
 
-	if usr, err := ks.getUser(args.Username); err == nil || usr != nil {
+	usr, err := ks.getUser(args.Username)
+	switch {
+	case err == nil || usr != nil:
 		return fmt.Errorf("user already exists: %s", args.Username)
+	case !usr.CheckPassword(args.Password):
+		return fmt.Errorf("incorrect password for user %q", args.Username)
```

Comment:
`usr` is invalid in this case, because the previous case makes sure the user doesn't exist.

I think what you meant to be doing here is checking to make sure that `userData.User.CheckPassword` returns true.

Reply:
Fixed. Standardized the keystore password error messages as an extra bonus.